src/shared/cpu-set-util.c

```diff
@@ -287,3 +295,85 @@ int cpu_set_from_dbus(const uint8_t *bits, size_t size, CPUSet *set) {
         s = (CPUSet) {};
         return 0;
 }
+
+static int numa_policy_to_mempolicy(const NUMAPolicy *policy, unsigned long *maxnode, unsigned long **nodes) {
+        unsigned node, bits = 0, ulong_bits;
+        _cleanup_free_ unsigned long *out = NULL;
+
+        assert(policy);
+        assert(maxnode);
+        assert(nodes);
+
+        /* Take a shortcut here if policy is set to default or local since kernel requires that we pass NULL as nodemask */
+        if (policy->type == NUMA_POLICY_DEFAULT || policy->type == NUMA_POLICY_LOCAL)
+                goto out;
+
+        bits = policy->nodes.allocated * 8;
+        ulong_bits = sizeof(unsigned long) * 8;
+
+
+        out = new0(unsigned long, DIV_ROUND_UP(policy->nodes.allocated, sizeof(unsigned long)));
+        if (!out)
+                return -ENOMEM;
+
+        for (node = 0; node < bits; node++)
+                if (CPU_ISSET_S(node, policy->nodes.allocated, policy->nodes.set))
+                        out[node / ulong_bits] |= 1ul << (node % ulong_bits);
+
+out:
+        *nodes = TAKE_PTR(out);
+        *maxnode = bits + 1;
+
+        return 0;
+}
+
+int apply_numa_policy(const NUMAPolicy *policy) {
+        int mode, r;
+        _cleanup_free_ unsigned long *nodes = NULL;
+        unsigned long maxnode;
+
+        assert(policy);
+
+        if (get_mempolicy(NULL, NULL, 0, 0, 0) < 0 && errno == ENOSYS)
```

Comment:
Hmm, why is this check needed?

Reply:
This is not strictly needed. Idea here is to have an "early return" on machines which don't support NUMA.

I mean conversion between parsed configuration and actual representation expected by ```set_mempolicy()``` calls ```malloc()```. I think it is better if we avoid it knowing that NUMA is not supported.

Reply:
I'm was wondering whether it's worth the extra syscall. But OK, seems reasonable to leave this in.